### DISCUSSION TITLE
Silent install + options + path?

### DISCUSSION THREAD

**Discussion by erasmion** [tnodir/fort#189] [Q&A] [answered]
Hello and thank you for this amazing program.   I guess it's clear in the subject:  is there a way to silent install FW choosing custom options (ie. portable) and/or a custom path?

**Comment by tnodir** [comment, marked as answer]
You can use `/SILENT` argument to install with default path and options. But you can't customize the path & options, because you can't customize them in update too.

Also you can unpack the installer e.g. to current directory by `/SILENT /UNPACK=.`. And setup a portable installation as you want.

**Reply by erasmion** [reply]
wow, thanks for such a quick reply!   

I think that is still a very good working situation, so on a fresh Windows I can have a batch file to:

1 unpack the installer
2 copy the unpacked files to the custom directory
3 make a shortcut link 
4 I guess I would also be able to set it as service (like when installing) using the following command (haven't tried yet):

`sc.exe create <FortFirewallSvc> binpath= "C:\_SHELLS\FORT FIREWALL\FortFirewall.exe" --service start= auto`

**Reply by tnodir** [reply]
You can use following cmd line arguments of installed `FortFiewall.exe`:
- `-i portable` - create `README.portable` file in app's dir.
- `-i service` - install a Windows Service
- `-i explorer` - integrate with Windows Explorer for current user
- `-u` - uninstall all above options

**Reply by erasmion** [reply]
that's great - will give it a try.  Thank you!
